sdk/ios/Configsum/Configuration/Environment.swift

```diff
@@ -23,18 +24,21 @@ public struct Environment {
     ///   - baseConfigurationName: name for the base configuration
     ///   - hostName: host endpoint
     ///   - port: optional attribute for http/https requests
+    ///   - urlScheme: The scheme subcomponent of the URL. if not set it defaults to https
     public init(log: Bool,
                 token: String,
                 headers: [String: [String]],
                 baseConfigurationName: String,
                 hostName: String,
-                port: Int? = nil) {
+                port: Int? = nil,
+                urlScheme: String = "https") {
```

Comment:
What happens if the caller provides an unsupported scheme? Should we make this an enum instead maybe?

Reply:
From the Apple docs:

`If you attempt to set the scheme to an invalid scheme string, this class throws an exception.`

Reply:
introducing a enum will break older clients